src/tools/pub/cmd/pub-upgrade.md

```diff
@@ -105,10 +105,55 @@ run `dart pub upgrade` again to upgrade to a later version.
 
 The `dart pub upgrade` command supports the
 [`dart pub get` options](/tools/pub/cmd/pub-get#options).
-
 For options that apply to all pub commands, see
 [Global options](/tools/pub/cmd#global-options).
 
+### `--dry-run` or `-n`
+
+Reports the dependencies that would be changed,
+but doesn't make the changes. This is useful if you
+want to analyze updates before making them.
+
+### `--major-versions`
+
+Gets the packages that [`dart pub outdated`][] lists as _resolvable_,
+ignoring any upper-bound constraint in the `pubspec.yaml` file.
+Also updates `pubspec.yaml` with the new constraints.
+
+[`dart pub outdated`]: /tools/pub/cmd/pub-outdated
+
+{{ site.alert.tip }}
+  Commit the `pubspec.yaml` file before running this command,
+  so that you can undo the changes if necessary.
+{{ site.alert.end }}
+To check which dependencies will be upgraded,
+you can use `dart pub upgrade --major-versions --dry-run`.
+
```

Comment:
@micpap25 also it's actually possible to do:
```
dart pub upgrade --major-versions foo bar ...
```
Assume you only want to upgrade major versions for packages `foo` and `bar`...

When running in this mode, it will still allow upgrades of other packages, but only if those upgrades don't require changes in `pubspec.yaml`. So in effect this will only do major version upgrades for packages `foo` and `bar`... other packages will get normal upgrade, if necessary.

--------

This partial behavior is also available for `--null-safety` with `dart pub upgrade --null-safety <package>...`, but here it's discouraged, because partial null-safety migration is a bad idea. So I don't think we should highlight the partial null-safety migration functionality.

Besides `dart pub upgrade --null-safety` will, if it fails, print a message saying which packages didn't have a null-safety compatible version, and then print the command for running a partial null-safety migration -- and discourage users from doing so :rofl: 

But if you really need to do partial null-safety migration, you'll discover it when you run `dart pub upgrade --null-safety` so let's not complicate documentation with such details.